### DISCUSSION TITLE
elasticsearch5  'actionGet' method can not be intercepted

### DISCUSSION THREAD

**Discussion by larrying** [apache/skywalking#8152] [General]
I faced a problem that elasticsearch5  'actionGet' method can not be intercepted, if i change it to 'get' method, it works. Is it bug?

**Comment by wu-sheng** [comment]
I don't understand what do you mean `can`? Is that a real method in the current class? `abstract` methods can't be intercepted, unless you use `HierarchyMatch` for the parent class.

**Reply by wu-sheng** [reply]
First of all, you should not post this graph. I can read the Chinese in your screenshot, this is actually a security leak from your company policy. Please delete it ASAP.

**Reply by larrying** [reply]
thank you for reminding me. I have deleted it 

Is it bug ？

**Reply by wu-sheng** [reply]
Then, the error is quite clear. The `PlainListenableActionFutureInterceptor` doesn't fit the target method invocation.

**Reply by wu-sheng** [reply]
More specifically, this is this line from your stack, https://github.com/apache/skywalking-java/blob/a7a4faabb158ea8322170e3c7cae0210a8c4e7b4/apm-sniffer/apm-sdk-plugin/elasticsearch-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/elasticsearch/v5/PlainListenableActionFutureInterceptor.java#L56.

The interceptor can't do `response.getHits().getTotalHits()` from JVM error report. I think this must be caused by different method parameter and return value(`SearchResponse response = (SearchResponse) ret`).

**Reply by larrying** [reply]
if i want to resolve this problem, enhance the PlainListenableActionFutureInterceptor?

**Reply by wu-sheng** [reply]
I don't know who wrote this plugin, I just explain what you asked about. SkyWalking's plugins are contributed by hundreds of people. 
You actually should basically understand how plugin works, then try to enhance. But enhancement doesn't have to relate to this interceptor. One plugin has multiple interceptors for different cases.
